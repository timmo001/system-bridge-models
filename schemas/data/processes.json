{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "description": "Processes",
  "properties": {
    "id": {
      "type": "string",
      "description": "Event ID"
    },
    "count": {
      "type": "number"
    },
    "processes": {
      "type": "array",
      "description": "Processes",
      "items": {
        "type": "object",
        "description": "Process",
        "properties": {
          "id": {
            "type": "string",
            "description": "ID"
          },
          "name": {
            "type": "string",
            "description": "Name"
          },
          "cpu_usage": {
            "type": "number",
            "description": "CPU usage percentage"
          },
          "created": {
            "type": "number",
            "description": "Created time (epoch)"
          },
          "memory_usage": {
            "type": "number",
            "description": "Memory usage percentage"
          },
          "path": {
            "type": "string",
            "description": "Path"
          },
          "status": {
            "type": "string",
            "description": "Status"
          },
          "username": {
            "type": "string",
            "description": "Username"
          },
          "working_directory": {
            "type": "string",
            "description": "Working directory"
          }
        },
        "required": ["id", "name"],
        "additionalProperties": true
      }
    },
    "last_updated": {
      "type": "object",
      "description": "Last updated",
      "properties": {
        "count": {
          "type": "number"
        },
        "processes": {
          "type": "number"
        }
      },
      "additionalItems": {
        "type": "number"
      },
      "required": ["count", "processes"]
    }
  },
  "additionalItems": {},
  "required": ["count", "processes"]
}
