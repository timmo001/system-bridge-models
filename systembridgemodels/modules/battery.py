"""Battery."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Battery:
    """Battery."""

    is_charging: bool | None = None
    percentage: float | None = None
