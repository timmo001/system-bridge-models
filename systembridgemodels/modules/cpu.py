"""CPU."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class CPUFrequency:
    """CPU Frequency."""

    current: float | None = None
    min: float | None = None
    max: float | None = None


@dataclass
class CPUStats:
    """CPU Stats."""

    ctx_switches: int | None = None
    interrupts: int | None = None
    soft_interrupts: int | None = None
    syscalls: int | None = None


@dataclass
class CPUTimes:
    """CPU Times."""

    user: float | None = None
    system: float | None = None
    idle: float | None = None
    interrupt: float | None = None
    dpc: float | None = None


@dataclass
class CPU:
    """CPU."""

    count: int | None = None
    frequency: CPUFrequency | None = None
    frequency_per_cpu: list[CPUFrequency] | None = None
    load_average: float | None = None
    power_package: float | None = None
    power_per_cpu: list[float] | None = None
    stats: CPUStats | None = None
    temperature: float | None = None
    times: CPUTimes | None = None
    times_per_cpu: list[CPUTimes] | None = None
    times_percent: CPUTimes | None = None
    times_percent_per_cpu: list[CPUTimes] | None = None
    usage: float | None = None
    usage_per_cpu: list[float] | None = None
    voltage: float | None = None
