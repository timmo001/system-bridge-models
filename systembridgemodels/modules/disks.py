"""Disks."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class DiskIOCounters:
    """Disk IO Counters."""

    read_count: int
    write_count: int
    read_bytes: int
    write_bytes: int
    read_time: int
    write_time: int


@dataclass
class DiskUsage:
    """Disk Usage."""

    total: int
    used: int
    free: int
    percent: float


@dataclass
class DiskPartition:
    """Disk Partition."""

    device: str
    mount_point: str
    filesystem_type: str
    options: str
    max_file_size: int
    max_path_length: int
    usage: DiskUsage | None = None


@dataclass
class Disk:
    """Disk."""

    name: str
    partitions: list[DiskPartition]
    io_counters: DiskIOCounters | None = None


@dataclass
class Disks:
    """Disks."""

    devices: list[Disk]
    io_counters: DiskIOCounters | None = None
