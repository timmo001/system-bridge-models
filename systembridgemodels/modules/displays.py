"""Displays."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Display:
    """Display."""

    name: str
    resolution_horizontal: int
    resolution_vertical: int
    x: int
    y: int
    width: int | None = None
    height: int | None = None
    is_primary: bool | None = None
    pixel_clock: float | None = None
    refresh_rate: float | None = None


type Displays = list[Display]
