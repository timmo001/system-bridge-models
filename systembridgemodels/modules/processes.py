"""Processes."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Process:
    """Process."""

    id: float
    name: str | None = None
    cpu_usage: float | None = None
    created: float | None = None
    memory_usage: float | None = None
    path: str | None = None
    status: str | None = None
    username: str | None = None
    working_directory: str | None = None


type Processes = list[Process]
