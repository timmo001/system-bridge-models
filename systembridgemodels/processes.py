# generated by datamodel-codegen:
#   filename:  processes.json

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Extra, Field


class Process(BaseModel):
    """
    Process
    """

    class Config:
        extra = Extra.allow

    id: str = Field(..., description="ID")
    name: str = Field(..., description="Name")
    cpu_usage: Optional[float] = Field(None, description="CPU usage percentage")
    created: Optional[float] = Field(None, description="Created time (epoch)")
    memory_usage: Optional[float] = Field(None, description="Memory usage percentage")
    path: Optional[str] = Field(None, description="Path")
    status: Optional[str] = Field(None, description="Status")
    username: Optional[str] = Field(None, description="Username")
    working_directory: Optional[str] = Field(None, description="Working directory")


class LastUpdated(BaseModel):
    """
    Last updated
    """

    count: float
    processes: float


class Processes(BaseModel):
    """
    Processes
    """

    id: Optional[str] = Field(None, description="Event ID")
    count: float
    processes: list[Process] = Field(..., description="Processes")
    last_updated: Optional[LastUpdated] = Field(None, description="Last updated")
