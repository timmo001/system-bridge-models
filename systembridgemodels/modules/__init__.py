"""Modules."""
from __future__ import annotations

from dataclasses import dataclass
from enum import Enum

from .battery import Battery
from .cpu import CPU
from .disks import Disks
from .displays import Displays
from .gpus import GPUs
from .media import Media
from .memory import Memory
from .networks import Networks
from .processes import Processes
from .sensors import Sensors
from .system import System


class DataEnum(Enum):
    """Data Enum."""

    BATTERY = "battery"
    CPU = "cpu"
    DISKS = "disks"
    DISPLAYS = "displays"
    GPUS = "gpus"
    MEDIA = "media"
    MEMORY = "memory"
    NETWORKS = "networks"
    PROCESSES = "processes"
    SENSORS = "sensor"
    SYSTEM = "system"


@dataclass
class GetData:
    """Get Data."""

    modules: list[str]


@dataclass
class RegisterDataListener(GetData):
    """Register Data Listener."""


@dataclass
class ModulesData:
    """Data."""

    battery: Battery | None = None
    cpu: CPU | None = None
    disks: Disks | None = None
    displays: Displays | None = None
    gpus: GPUs | None = None
    media: Media | None = None
    memory: Memory | None = None
    networks: Networks | None = None
    processes: Processes | None = None
    sensors: Sensors | None = None
    system: System | None = None
