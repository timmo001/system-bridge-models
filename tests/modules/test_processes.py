"""Test the networks module model."""

from syrupy.assertion import SnapshotAssertion

from systembridgemodels.fixtures.modules.processes import FIXTURE_PROCESSES
from systembridgemodels.modules.processes import Process


def test_networks(snapshot: SnapshotAssertion):
    """Test the processes model."""
    processes = FIXTURE_PROCESSES
    assert isinstance(processes, list)
    assert isinstance(processes[0], Process)
    assert processes == snapshot
