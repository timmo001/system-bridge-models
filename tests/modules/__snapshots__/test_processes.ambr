# serializer version: 1
# name: test_networks
  list([
    Process(id=1234, name='name', cpu_usage=12.3, created=12.3, memory_usage=12.3, path='/path', status='running', username='username', working_directory='/working/directory'),
  ])
# ---
